tutormfe/plugin.py

```diff
@@ -58,6 +58,25 @@
 )
 
 
+@hooks.Filters.COMPOSE_MOUNTS.add()
+def _mount_frontend_apps(volumes, name):
+    """
+    If the user mounts any repo named frontend-app-APPNAME, then make sure
+    it's available in the APPNAME service container. This is only applicable
+    in dev mode, because in production, all MFEs are built and hosted on the
+    singular 'mfe' service container.
+    """
+    prefix = "frontend-app-"
+    if name.startswith(prefix):
+        # Assumption:
+        # For each repo named frontend-app-APPNAME, there is an associated
+        # docker-compose service named APPNAME. If this assumption is broken,
+        # then Tutor will try to mount the repo in a service that doesn't exist.
```

Comment:
I am torn on this and would appreciate your input @regisb : 

Currently, I allow `--mount` to be specified for any arbitrary `frontend-app-` repo. This is nice, because it would allow a user who has configured a custom MFE (say, frontend-app-authn) to bind-mount it without any hassle. The downside, though, is that we get a fairly confusing error message if the custom MFE hasn't actually been configured, since the custom MFE's service would be missing from env/dev/docker-compose.yml:
```
$ tutor dev start -d -m ../frontend-app-learning/ -m ../frontend-app-authn/
Bind-mount: /home/kyle/openedx/frontend-app-learning -> /openedx/app in learning
Bind-mount: /home/kyle/openedx/frontend-app-authn -> /openedx/app in authn
docker-compose -f /home/kyle/.local/share/tutor/env/local/docker-compose.yml -f /home/kyle/.local/share/tutor/env/dev/docker-compose.yml -f /home/kyle/.local/share/tutor/env/local/docker-compose.tmp.yml --project-name tutor_dev up --remove-orphans --build -d
ERROR: The Compose file is invalid because:
Service authn has neither an image nor a build context specified. At least one must be provided.
```

Options I see:
* Keep it as-is.
* Keep it as-is, but try to improve the error message in Tutor.
* Change it to a hard-coded allowlist of bindmounts (for now: profile, account, learning, gradebook).
* Add a `config` argument to COMPOSE_MOUNTS, which would allow us to see which MFE have been configured.

Reply:
I think that your approach is sensible. It's true that the error message is difficult to understand, but we expect that users who attempt to bind-mount volumes are rather advanced. Maybe we can simply add a note to the README?

I'm more concerned by the fact that this bind-mount will not work in `tutor local`, because the container is then named "mfe" and not `app_name`. Maybe we should add another note about this in the README.

Reply:
Good points. Done.